src/gui/wizard/owncloudwizardcommon.cpp

```diff
@@ -68,6 +72,77 @@ namespace WizardCommon {
         errorLabel->setVisible(false);
     }
 
+    void customizeSpinBoxStyle(QSpinBox *spinBox)
+    {
+        auto spinBoxPalette = spinBox->palette();
+#ifdef Q_OS_WIN
+        // Windows always uses a SpinBox with a white background but we change the color of the text so we need
+        // to change the text color here
+        spinBoxPalette.setColor(QPalette::Text, Qt::black);
+        QColor colorTextDisabled(spinBoxPalette.color(QPalette::Text));
+        colorTextDisabled.setAlpha(128);
+        spinBoxPalette.setColor(QPalette::Disabled, QPalette::Text, colorTextDisabled);
+#endif
+#ifdef Q_OS_LINUX
+
+        spinBoxPalette.setColor(QPalette::WindowText, Theme::instance()->wizardHeaderBackgroundColor());
+        QColor colorWindowTextDisabled(spinBoxPalette.color(QPalette::WindowText));
+        colorWindowTextDisabled.setAlpha(128);
+        spinBoxPalette.setColor(QPalette::Disabled, QPalette::WindowText, colorWindowTextDisabled);
+#endif
+        spinBox->setPalette(spinBoxPalette);
+    }
+
+    void customizeSecondaryButtonStyle(QAbstractButton *button)
+    {
+#ifdef Q_OS_LINUX
+        // Only style push buttons on Linux as on other platforms we are unable to style the background color
```

Comment:
Yes, in the past we used style sheets to make the buttons blue. But if we use style sheets, the buttons do not look native anymore as can be seen on the current wizard in OSX and Windows.

See also the warning here: https://doc.qt.io/qt-5/qpalette.html#details 
I guess this warning applies to Windows 10 as well but really if you know how to change the background color of push buttons with the palette on Windows and OSX I would love to hear that:) I searched quite a while for a solution...

Reply:
> And I know that designers might not like that as an answer they don't get 100% the style expected but with widgets based there's a price to the desktop integration it's that you need to abandon some freedom to the platform. This is kind of intended.

I'm with you with that :)

The problem was that when you, for example, set the font to white, you get also a white font on the buttons (If I remember correctly) but the background color of the button does not change to your expected background color (blue for example). Now you can't read the text on the buttons anymore. This is especially an issue for us because we allow our customers to brand their application. Now when they choose a dark background and a light font color they don't see the text anymore on the buttons. 

In my opinion, we should stay away from modifying the palette. We would have a lot less trouble, and the application would look native (I would argue that users love that).

Reply:
@er-vin I removed the ifdefs as far as possible